src/Slider/_tests_/index.test.js

```diff
@@ -0,0 +1,488 @@
+        60,
+        70,
+        80,
+        90,
+        100,
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+      ]}
+    />,
+  );
+  const instance = component.instance();
+  test("calculateValueFromPosition should return proper value", () => {
+    expect(instance.calculateValueFromPosition(52)).toEqual(7);
+    expect(instance.calculateValueFromPosition(150)).toEqual(31);
+    instance.setState({ value: [15, 12], handleIndex: 0 });
+    expect(instance.calculateValueFromPosition(140)).toEqual(28);
+    instance.setState({ handleIndex: null });
+    expect(instance.calculateValueFromPosition(140)).toEqual(29);
+    instance.setState({ value: [12, 15], handleIndex: 0 });
+    expect(instance.calculateValueFromPosition(140)).toEqual(29);
+    instance.setState({ handleIndex: null });
+    expect(instance.calculateValueFromPosition(140, true)).toEqual(28);
+    instance.setState({ handleIndex: 1 });
+    expect(instance.calculateValueFromPosition(140, true)).toEqual(28);
+  });
+  test("should have Hide component", () => {
+    expect(component.find("Hide").exists()).toBe(true);
+  });
+  test("moveValueByStep should return proper value", () => {
+    instance.setState({ handleIndex: 0 });
+    expect(instance.moveValueByStep(1)).toEqual([13, 15]);
+    expect(instance.moveValueByStep(-1)).toEqual([11, 15]);
+    expect(instance.moveValueByStep(3)).toEqual([15, 15]);
+    expect(instance.moveValueByStep(-3)).toEqual([9, 15]);
+    expect(instance.moveValueByStep(-15)).toEqual([1, 15]);
+    expect(instance.moveValueByStep(100)).toEqual([100, 15]);
+  });
+  test("defaultValue changes the value state", () => {
+    component.setProps({ defaultValue: [1, 100] });
+    expect(instance.state.value).toEqual([1, 100]);
+  });
+  it("should execute keyDown event with arrow up", () => {
+    instance.setState({ handleIndex: 0, value: [50, 60] });
+    instance.handleKeyDown({
+      stopPropagation: () => {},
+      preventDefault: () => {},
+      keyCode: 38,
```

Comment:
maybe you could import the keys from `keyMaps.js` for better readability of the test

Reply:
it also looks like this part is repeating quite a lot, so it could be more terse to break it out into a helper function 

```js
// somewhere outside of `describe` and `test`
function testKeyTriggeringEvent(instance, expect, initialState, keyCode, stateValue, onHandler) {
    instance.setState(initialState);
    instance.handleKeyDown({
      stopPropagation: () => {},
      preventDefault: () => {},
      keyCode,
    });
    expect(instance.state.value).toEqual(stateValue);
    expect(onHandler).toHaveBeenCalled();
}

// later in the test code

it("should execute keyDown event with arrow up", () =>
    testKeyTriggeringEvent(instance, expect, { handleIndex: 0, value: [50, 60] }, keyMaps.ARROW_UP, [51, 60], onChange)
);
```